openedx_lti_tool_plugin/views.py

```diff
@@ -429,11 +430,21 @@ def handle_ags(
         if AGS_SCORE_SCOPE not in ags_endpoint.get('scope', []):
             raise LtiToolLaunchException(_(f'Missing required AGS scope: {AGS_SCORE_SCOPE}'))
 
-        return LtiGradedResource.objects.get_or_create(
-            lti_profile=lti_profile,
-            context_key=context_key,
-            lineitem=lineitem,
-        )
+        graded_resource_values = {
+            'lti_profile': lti_profile,
+            'context_key': context_key,
+            'lineitem': lineitem,
+        }
+
+        try:
```

Comment:
I guess that using `get_or_create` would simplify the logic here, but I suppose that you want to make sure that .full_clean is called. That correct?. @alexjmpb

Reply:
Yes correct, with get_or_create the field validators are not performed

Reply:
Have you though about overriding the model's save method to make sure that your validations are always called?. @alexjmpb

Reply:
Unless you are 100% sure there isn't going to be other way to create those objects, then It would be fine for me. But if in future any other creation of such objects should also call full_clean. @alexjmpb